pkg/cmd/bootstrap.go

```diff
@@ -283,26 +289,33 @@ func checkBootstrapDependencies(io *BootstrapParameters, client *utility.Client,
 	// in case custom Sealed Secrets namespace/service name are provided, try them first
 	// We do not add Sealed Secret Operator to missingDeps since we this dependency can be resolved
 	// by optional flags or interactive user inputs.
-	if (io.BootstrapOptions.SealedSecretsService.Namespace != "" && io.BootstrapOptions.SealedSecretsService.Namespace != defaultSealedSecretsServiceName.Namespace) ||
-		(io.BootstrapOptions.SealedSecretsService.Name != "" && (io.BootstrapOptions.SealedSecretsService.Name != defaultSealedSecretsServiceName.Name)) {
+	if !io.Insecure && ((io.BootstrapOptions.SealedSecretsService.Namespace != "" && io.BootstrapOptions.SealedSecretsService.Namespace != defaultSealedSecretsServiceName.Namespace) ||
+		(io.BootstrapOptions.SealedSecretsService.Name != "" && (io.BootstrapOptions.SealedSecretsService.Name != defaultSealedSecretsServiceName.Name))) {
 
 		spinner.Start("Checking if Sealed Secrets is installed with custom configuration", false)
 		if err := checkAndSetSealedSecretsConfig(io, client, io.BootstrapOptions.SealedSecretsService); err != nil {
 
 			warnIfNotFound(spinner, "Provided Sealed Secrets namespace/name are not valid. Please verify", err)
+			io.Insecure = true
 			if !apierrors.IsNotFound(err) {
 				return fmt.Errorf("failed to check for Sealed Secrets Operator: %w", err)
 			}
+			utility.DisplayUnsealedSecretsWarning()
 		}
 	} else {
-		// use default configuration to interact with Sealed Secrets
-
-		spinner.Start("Checking if Sealed Secrets is installed with the default configuration", false)
-		if err := checkAndSetSealedSecretsConfig(io, client, defaultSealedSecretsServiceName); err != nil {
-
-			warnIfNotFound(spinner, "Please install Sealed Secrets Operator from OperatorHub", err)
-			if !apierrors.IsNotFound(err) {
-				return fmt.Errorf("failed to check for Sealed Secrets Operator: %w", err)
+		if io.Insecure {
+			utility.DisplayUnsealedSecretsWarning()
+		} else {
+			spinner.Start("Checking if Sealed Secrets is installed with the default configuration", false)
+			err := checkAndSetSealedSecretsConfig(io, client, defaultSealedSecretsServiceName)
+			if err != nil {
+				// use default configuration to interact with Sealed Secrets
+				warnIfNotFound(spinner, "The Sealed Secrets Operator was not detected", err)
+				io.Insecure = true
```

Comment:
In this method where we check for dependencies, it applies to both non-interactive and interactive mode.  So we can't prompt the user in that method, or else we will 'break' the automated tests.

If you specified `--sealed-secrets-ns` and `--sealed-secrets-svc` from the command line, then this does allow the user to provide the custom, non-default configuration.  However, it's not prompted 'later' during the interactive mode session.  That is one way that users can still do scenario 2.  I think for scenario 1, users can just specify the --insecure flag.

But having said that, yes, during interactive mode there is no prompt because of the assumption. So, I've added an additional prompt to basically prompt the user for it, and it will NOT continuously prompt the user until there is a  valid custom namespace and name, like how the other typical prompts behave with validation.  Here is an example:. Note the second question in the interactive mode:


```
Checking dependencies

 ✗  Checking if Sealed Secrets is installed with the default configuration [11s] [The Sealed Secrets Operator was not detected]
    WARNING: Unencrypted secrets will be created in a secrets folder that is a sibling to the designated output folder
             Deploying this GitOps configuration without encrypting secrets is insecure and is not recommended
 ✓  Checking if ArgoCD is installed with the default configuration [12s]                                                                      
 ✓  Checking if OpenShift Pipelines Operator is installed with the default configuration [12s]

Starting interactive prompt

? Do you want to accept all default values and be prompted only for the minimum required options? no
? The Sealed Secrets operator with the default configuration has not been found, do you want to provide a custom namespace? yes
? Provide a namespace in which the Sealed Secrets operator is installed, automatically generated secrets are encrypted with this operator? cicd
 ✗  The namespace cicd is not found on the cluster
  WARNING: The Sealed Secrets Operator is not found with the provided namespace/name. Unsealed secrets will be generated instead
? Provide the URL for your GitOps repository [? for help] 
```


And

```
Starting interactive prompt

? Do you want to accept all default values and be prompted only for the minimum required options? no
? The Sealed Secrets operator with the default configuration has not been found. Do you want to enter a custom namespace? yes
? Provide a namespace in which the Sealed Secrets operator is installed, automatically generated secrets are encrypted with this operator? openshift-operators
? Name of the Sealed Secrets Service that encrypts secrets test
 ✗  services "test" not found
  WARNING: The Sealed Secrets Operator is not found with the provided namespace/name. Unsealed secrets will be generated instead
? Provide the URL for your GitOps repository [? for help] 
```